File: 백준/Silver/1966. 프린터 큐/프린터 큐.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;

public class Main {
  public static void main(String[] args) throws IOException {
    BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
    int total = Integer.parseInt(bufferedReader.readLine());
    for(int i = 0; i< total; i++){
      String[] strings = bufferedReader.readLine().split(" ");
      int M = Integer.parseInt(strings[1]);

      String[] importance = bufferedReader.readLine().split(" ");
      Deque<Integer> queue = new ArrayDeque<>();
      Deque<Integer> queue2 = new ArrayDeque<>();
      for(int j = 0; j < importance.length; j++){
        queue.add(Integer.parseInt(importance[j]));
        queue2.add(j);
      }
      int count =0;
      while(!queue.isEmpty()){
        Integer max = queue.stream().max(Integer::compareTo).orElse(0);
        if(queue.getFirst() >= max){
          if(queue2.getFirst() == M){
            count +=1;
            System.out.println(count);
            break;
          }
          queue.removeFirst();
          queue2.removeFirst();
          count++;

        } else{
          Integer first = queue.getFirst();
          queue.addLast(first);
          queue.removeFirst();
          Integer second = queue2.getFirst();
          queue2.addLast(second);
          queue2.removeFirst();
        }
      }
    }
  }
}

File: 백준/Silver/1966. 프린터 큐/프린터 큐.py
import sys
from collections import deque

## Input받기
## 큐에 저장한다
## 큐에 저장할때 최대값을 저장한다
## 큐의 첫번째값이 최대값이랑 같다면 출력 아니라면 넘긴다

T = int(sys.stdin.readline())
result = []

for _ in range(T):
    N, index = map(int, input().split())
    queue = deque(list(map(int, input().split())))
    count = 0
    cursor = index

    while(len(queue) > 0):
        maxValue = max(queue)
        currentValue = queue.popleft()
        if(currentValue < maxValue):
            queue.append(currentValue)
        elif (currentValue >= maxValue and cursor == 0):
            result.append(str(count+1))
            break
        else:
            count +=1

        if cursor == 0:
            cursor = len(queue) -1
        else:
            cursor -= 1

print("\n".join(result))
